src/goose/toolkit/screen.py

```diff
@@ -1,24 +1,54 @@
+import os
+import re
 import subprocess
 import uuid
 
+from rich.markdown import Markdown
+from rich.panel import Panel
+
 from goose.toolkit.base import Toolkit, tool
 
 
 class Screen(Toolkit):
     """Provides an instructions on when and how to work with screenshots"""
 
     @tool
-    def take_screenshot(self) -> str:
+    def take_screenshot(self, display: int = 1, resize: bool = False, target_size: int = 4) -> str:
         """
         Take a screenshot to assist the user in debugging or designing an app. They may need help with moving a screen element, or interacting in some way where you could do with seeing the screen.
+        Optionally, resize the image using the `sips` cli tool. You may do this fs there is potential for error in the payload size.
```

Comment:
i think at the moment, the error around the payload size is not surfaced to the model but instead exits the loop? we'll need an update in exchange i think for it to see the need to do this.

We might instead consider always resizing to 728x on the low side, given the openai approach. We can also switch from png to jpg.
> detail: high images are first scaled to fit within a 2048 x 2048 square, maintaining their aspect ratio. Then, they are scaled such that the shortest side of the image is 768px long.

That'd make for maybe 200kb uploads, so always well below the limit without needing the model to consider when it needs to resize